db/background_mgr.go

```diff
@@ -52,47 +76,160 @@ type BackgroundManagerStatus struct {
 // BackgroundManagerProcessI is an interface satisfied by any of the background processes
 // Examples of this: ReSync, Compaction
 type BackgroundManagerProcessI interface {
+	Init(clusterStatus []byte) error
 	Run(options map[string]interface{}, terminator chan struct{}) error
 	GetProcessStatus(status BackgroundManagerStatus) ([]byte, error)
 	ResetStatus()
 }
 
 func (b *BackgroundManager) Start(options map[string]interface{}) error {
+	b.terminator = make(chan struct{}, 1)
+	b.terminatorClosed = false
+
 	err := b.markStart()
 	if err != nil {
 		return err
 	}
 
+	var processClusterStatus []byte
+	if b.clusterAwareOptions != nil {
+		processClusterStatus, _, err = b.clusterAwareOptions.bucket.GetRaw(b.clusterAwareOptions.StatusDocID())
+		if err != nil && !base.IsDocNotFoundError(err) {
+			return err
+		}
+	}
+
 	b.resetStatus()
+
+	err = b.Process.Init(processClusterStatus)
+	if err != nil {
+		return err
+	}
+
+	if b.clusterAwareOptions != nil {
+		go func() {
+			ticker := time.NewTicker(time.Second)
+			for {
+				select {
+				case <-ticker.C:
+					// Implement a retry on here. It's not the end of the world if we miss a couple of these, but we
+					// want to give it a best attempt
+					err = b.UpdateStatusClusterAware()
+					if err != nil {
+						base.Errorf("Failed to update background manager status: %v", err)
```

Comment:
Errorf is usually reserved for severe errors that the user should act on.  This doesn't seem like the case here - warning seems sufficient, as the only loss of functionality here is the ability to resume.